micropython/ubirch/ubirch_sim.py

```diff
@@ -196,13 +200,19 @@ def _select_ss_entry(self, entry_id: str) -> str:
             print('found entry ID: ' + repr(self._decode_tag(data)))
         return code
 
-    def select(self):
+    def get_imsi(self) -> str:
         """
-        Select the SIM application to execute secure operations.
+        Get the international mobile subscriber identity (IMSI) from SIM
         """
         self.lte.pppsuspend()
-        self._execute(STK_APP_SELECT.format(APP_DF))
+        at_cmd = "AT+CIMI"
+        if self.DEBUG: print("++ " + at_cmd)
+        result = [k for k in self.lte.send_at_cmd(at_cmd).split('\r\n') if len(k.strip()) > 0]
+        if self.DEBUG: print('-- ' + '\r\n-- '.join([r for r in result]))
         self.lte.pppresume()
+        if result[-1] == 'OK':
+            return result[0]
```

Comment:
Possibly also check imsi is sane (maximum length and that it only contains numbers)? We might get almost anything from the SIM if something went wrong in the previous command/communication.

Reply:
added todo